packages/cosmic-swingset/lib/ag-solo/vats/nameHub.js

```diff
@@ -0,0 +1,82 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { makeStore } from '@agoric/store';
+
+// import './types.js';
+
+/**
+ * @returns {NameHubKit}
+ */
+export const makeNameHubKit = () => {
+  /** @typedef {Partial<PromiseRecord<unknown> & { value: unknown }>} NameRecord */
+  /** @type {Store<unknown, NameRecord>} */
+  const keyToRecord = makeStore('nameKey');
+
+  /** @type {NameHub} */
+  const nameHub = {
+    async lookup(...path) {
```

Comment:
Hmm, do we have tests for the path functionality? If not, maybe let's throw in that case with a message that it hasn't been implemented (tested) yet.

Reply:
The cases for 0, 1, 2 path elements are straightforward enough, so I will add a tests.